xarray/backends/netCDF4_.py

```diff
@@ -463,6 +463,15 @@ def open(
         if isinstance(filename, os.PathLike):
             filename = os.fspath(filename)
 
+        # Replace DAP protocol prefixes with https:// - netCDF4 library can't handle them
+        # These prefixes may be added by users to explicitly indicate DAP protocol
+        # Following pydap's convention, we convert to https://
+        # See: https://github.com/pydap/pydap/blob/0a2b0892611abaf0a9762ffd4f2f082cb8e497c2/src/pydap/handlers/dap.py#L103-L107
+        if isinstance(filename, str):
+            filename_lower = filename.lower()
+            if filename_lower.startswith(("dap2://", "dap4://")):
+                filename = "https://" + filename[7:]
+
```

Comment:
note for review. A sneaky thing this PR does is allow the `dap4://` syntax to work with `engine="netcdf4"` This makes it cleaner to ahve one dap url detection function, but may have downsides I'm unaware of.